### PR TITLE
Public iteam stuff

### PR DESCRIPTION
**Pull request by mlsteele**
This PR is like https://github.com/keybase/client/pull/8178 but with the iTeam IdentifyLite part stripped out. It adds public team awareness to the player, tweaks some iteam name formatting, and improves reader support.

### CODE REVIEW

File: go/teams/implicit.go [hunk 1]
```diff
@@ -93,20 +96,46 @@ func LookupOrCreateImplicitTeam(ctx context.Context, g *libkb.GlobalContext, nam
...
+	var readerNames []string
+	for _, u := range impTeamName.Readers.KeybaseUsers {
+		readerNames = append(readerNames, u)
+	}
+	for _, u := range impTeamName.Readers.UnresolvedUsers {
+		readerNames = append(readerNames, u.String())
 	}
-	sort.Slice(names, func(i, j int) bool {
-		return names[i] < names[j]
+	sort.Slice(readerNames, func(i, j int) bool {
```
Comment (maxtaco) [inline go/teams/implicit.go]: I think `sort.Strings` can work here

File: go/teams/implicit.go [hunk 2]
```diff
@@ -93,20 +96,46 @@ func LookupOrCreateImplicitTeam(ctx context.Context, g *libkb.GlobalContext, nam
...
+	var writerNames []string
 	for _, u := range impTeamName.Writers.KeybaseUsers {
-		names = append(names, u)
+		writerNames = append(writerNames, u)
 	}
 	for _, u := range impTeamName.Writers.UnresolvedUsers {
-		names = append(names, u.String())
+		writerNames = append(writerNames, u.String())
+	}
+	sort.Slice(writerNames, func(i, j int) bool {
```
Comment (maxtaco) [inline go/teams/implicit.go]: `sort.Strings`

File: go/teams/implicit.go [hunk 3]
```diff
@@ -93,20 +96,46 @@ func LookupOrCreateImplicitTeam(ctx context.Context, g *libkb.GlobalContext, nam
 }
 
 func FormatImplicitTeamName(ctx context.Context, g *libkb.GlobalContext, impTeamName keybase1.ImplicitTeamName) (string, error) {
-	var names []string
+	var writerNames []string
 	for _, u := range impTeamName.Writers.KeybaseUsers {
-		names = append(names, u)
+		writerNames = append(writerNames, u)
```
Comment (maxtaco) [inline go/teams/implicit.go]: are these `toLower`ed?

**Comment by keybase-ci-visdiff** [comment]
The commits d12af463c7cb...972c59cedb2a introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503429850624-d12af463c7cb-972c59cedb2a-linux/Markdown-emoji-d12af463c7cb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503429850624-d12af463c7cb-972c59cedb2a-linux/Markdown-emoji-972c59cedb2a.png)  
   Markdown-emoji comparison
</summary>

File: go/teams/implicit.go [hunk 4]
```diff
@@ -93,20 +96,46 @@ func LookupOrCreateImplicitTeam(ctx context.Context, g *libkb.GlobalContext, nam
...
+	prefix := "private/"
+	if !impTeamName.IsPrivate {
+		prefix = "public/"
+	}
+	return prefix + normalized, nil
+}
+
+// time -> "2017-08-22"
+func formatConflictTime(thyme time.Time) string {
+	year, month, day := thyme.UTC().Date()
```
Comment (maxtaco) [inline go/teams/implicit.go]: I feel like I've written this before in go, maybe in my now-dead PR.

`time.Format`

File: go/teams/implicit.go (hunk 3 above)
Comment (mlsteele) [inline go/teams/implicit.go]: You think they should be here?

**Pull request merged**